fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/query/FDBRecordStoreQueryTest.java

```diff
@@ -295,10 +294,14 @@ public void queryWithContinuation() throws Exception {
                     .setFilter(Query.field("num_value_2").equalsValue(0))
                     .build();
             plan = planner.plan(query);
-            assertThat(plan, filter(query.getFilter(), typeFilter(anything(), scan(unbounded()))));
-            assertEquals(913370522, plan.planHash(PlanHashable.PlanHashKind.LEGACY));
+            assertThat(plan, filter(Objects.requireNonNull(query.getFilter()), typeFilter(anything(), scan(unbounded()))));
+            if (planner instanceof RecordQueryPlanner) {
+                assertEquals(913370522, plan.planHash(PlanHashable.PlanHashKind.LEGACY));
             // TODO: https://github.com/FoundationDB/fdb-record-layer/issues/1074
             // assertEquals(389700036, plan.planHash(PlanHashable.PlanHashKind.STRUCTURAL_WITHOUT_LITERALS));
+            } else {
+                assertEquals(-1244637277, plan.planHash(PlanHashable.PlanHashKind.LEGACY));
```

Comment:
Maybe these call for another comment on the general nature of the difference with the new planner.

Reply:
The plan can be different as it may use `QueryComponent` in filters rather than `QueryPredicate`s, also sometimes predicates are reordered or the plan is different but verifiably better.